pool/chainstate.go

```diff
@@ -119,7 +120,18 @@ func (cs *ChainState) pruneAcceptedWork(ctx context.Context, height uint32) erro
 		}
 		confs, err := cs.cfg.GetBlockConfirmations(ctx, hash)
 		if err != nil {
-			return err
+			var rErr *dcrjson.RPCError
+			switch errors.As(err, &rErr) {
```

Comment:
This does not work. I suspect because `err` here is not the error which has been returned by the RPC client, but instead it is an error we have created ourselves in `hub.go`

https://github.com/decred/dcrpool/blob/7da5b4b51d7c69546b270591d18982af0e457faf/pool/hub.go#L450-L453